### PR TITLE
【paddle.fleet】add document on localsgd in DistributedStrategy

### PR DESCRIPTION
**Pull request by gavin1332**
### PR types
Others

### PR changes
Docs

### Describe
add comments for localsgd strategy

### CODE REVIEW

File: python/paddle/distributed/fleet/base/distributed_strategy.py [hunk 1]
```diff
@@ -634,6 +649,26 @@ def localsgd(self, flag):
...
     @property
     def localsgd_configs(self):
+        """
+        Set LocalSGD training configurations. LocalSGD has a configurable
+        setting that can be configured through a dict.
+
+        **Notes**:
+            **k_steps(int)**: The local steps for training before parameter
+                synchronization. Default 1. If strategy.auto is set True, the
+                local steps will be calculated automatically during training.
```
Comment (guru4elephant) [inline python/paddle/distributed/fleet/base/distributed_strategy.py:658]: I think currently k_steps is hard to decide when auto is set. Is there a heuristic that we can set the k_steps automatically.

**Review by guru4elephant — DISMISSED** [review]
LGTM

File: python/paddle/distributed/fleet/base/distributed_strategy.py (hunk 1 above)
Comment (gavin1332) [reply python/paddle/distributed/fleet/base/distributed_strategy.py:658]: [Adaptive Communication Strategies to Achieve the Best Error-Runtime Trade-off in Local-Update SGD](https://arxiv.org/pdf/1810.08313.pdf)
we have shown the reference. I will move it here

Comment (gavin1332) [reply python/paddle/distributed/fleet/base/distributed_strategy.py:658]: done

**Review by guru4elephant — APPROVED** [review]
LGTM

**Pull request merged**